xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitleTagSami.cpp

```diff
@@ -281,3 +305,12 @@ void CDVDSubtitleTagSami::LoadHead(CDVDSubtitleStream* samiStream)
     }
   }
 }
+
+void CDVDSubtitleTagSami::LoadColors()
```

Comment:
Is this really needed? The `system/colors.xml` file seems to be already loaded when the colormanager is initialized, it seems odd we need to parse the file again in `ConvertLine` in a subtitle tagger and store a copy of the color vector in another component (memory duplication). The colormanager seems to provide a `GetColor` method. Can't it be used instead?
Furthermore the colormanager currently belongs to the GUI scope. I don't think it belongs there as already mentioned in some other comment (I see colormanager and fontmanager as some sort of global resources available even if kodi is headless). Without that first refactoring you're introducing GUI includes into subtitles and inherently to videoplayer as well...
I don't doubt this fixes the issue at hand but sometimes the quick fix of an issue is not worth it if it makes it harder to refactor the core in the future. Sometimes we need to take a step back to do multiple steps forward. I think in this case we need to get the foundations right first